src/data/abilities/ab-attrs/prevent-bypass-speed-chance-ab-attr.ts

```diff
@@ -15,19 +15,20 @@ export class PreventBypassSpeedChanceAbAttr extends AbAttr {
   private readonly condition: (pokemon: Pokemon, move: Move) => boolean;
 
   constructor(condition: (pokemon: Pokemon, move: Move) => boolean) {
-    super(true);
+    super();
     this._flags.add(AbAttrFlag.PREVENT_BYPASS_SPEED_CHANCE);
     this.condition = condition;
   }
 
-  public override apply(pokemon: Pokemon, _simulated: boolean, cancelled: BooleanHolder): boolean {
+  public override apply(_pokemon: Pokemon, _simulated: boolean, cancelled: ValueHolder<boolean>): void {
+    cancelled.value = true;
+  }
+
+  public override canApply(...[pokemon]: Parameters<this["apply"]>): boolean {
     const turnCommand = globalScene.currentBattle.turnManager.findCommandFromPokemon(pokemon);
     const isCommandFight = turnCommand?.command === BattleCommand.FIGHT;
     const move = turnCommand?.turnMove?.move;
-    if (move && this.condition(pokemon, move) && isCommandFight) {
-      cancelled.value = true;
-      return true;
-    }
-    return false;
+
+    return move != null && this.condition(pokemon, move) && isCommandFight;
```

Comment:
Another micro-optimization.
```suggestion
    return move != null && isCommandFight && this.condition(pokemon, move);
```